metaci/repository/views.py

```diff
@@ -210,4 +190,66 @@ def github_push_webhook(request):
                 build.release_relationship_type = "test"
             build.save()
 
-    return HttpResponse("OK")
+
+def get_repository(push_payload):
+    repo_id = push_payload["repository"]["id"]
+    return Repository.objects.get(github_id=repo_id)
+
+
+def is_tag(ref):
+    """Returns true if ref corresponds to a tag. False otherwise"""
+    return True if ref.startswith(TAG_BRANCH_PREFIX) else False
+
+
+def get_or_create_branch(push, repo):
+    branch_name = get_branch_name_from_payload(push)
+
+    if branch_name:
+        branch, _ = Branch.objects.get_or_create(repo=repo, name=branch_name)
+        if branch.is_removed:
+            # resurrect the soft deleted branch
+            branch.is_removed = False
+            branch.save()
+
+    return branch
+
+
+def get_branch_name_from_payload(push):
+    branch_ref = push.get("ref")
+    if not branch_ref:
+        return HttpResponse("No branch found")
```

Comment:
@davisagli - Wondering what the best way to handle this case would be. We want this to be detected at the level of `github_push_webhook()` but this check is being performed a couple layers down from it. Not sure what the best way to go about this would be.